src/Analysis/Codelyzer.Analysis/CodeGraph.cs

```diff
@@ -0,0 +1,198 @@
+                invocationExpressions.ForEach(invocation => {
+                    var targetNode = graph.FirstOrDefault((n) => n.Identifier == invocation.SemanticFullClassType);
+
+                    //Skip methods in same class
+                    if (targetNode.Identifier != currentNode.Identifier)
+                    {
+                        var edge = new Edge()
+                        {
+                            EdgeType = invocation is ObjectCreationExpression ? EdgeType.ObjectCreation : EdgeType.Invocation,
+                            TargetNode = targetNode,
+                            Identifier = invocation.MethodName,
+                            SourceNode = currentNode
+                        };
+                        currentNode.Edges.Add(edge);
+                        targetNode.Edges.Add(edge);
+                    }
+                });
+
+                // Members (Properties)
+                var memberExpressions = classDeclaration.AllMemberAccessExpressions().Where(i => allNamespaces.Contains(i.Reference.Namespace)).ToList();
+                memberExpressions.ForEach(memberExpression => {
+                    var targetNode = graph.FirstOrDefault((n) => n.Identifier == memberExpression.SemanticFullClassType);
+
+                    //Skip methods in same class
+                    if (targetNode.Identifier != currentNode.Identifier)
+                    {
+                        var edge = new Edge()
+                        {
+                            EdgeType = EdgeType.MemberAccess,
+                            TargetNode = targetNode,
+                            Identifier = memberExpression.Identifier,
+                            SourceNode = currentNode
+                        };
+                        currentNode.Edges.Add(edge);
+                        targetNode.Edges.Add(edge);
+                    }
+                });
+
+            }
+
+            return graph;
+        }
+    }
+
+    public class Node
+    {
+        public Node()
+        {
+            Properties = new Dictionary<string, string>();
+            Edges = new List<Edge>();
+        }
+        public string Name { get; set; }
+        public string Identifier { get; set; }
+        public NodeType NodeType { get; set; }
+        public List<Edge> Edges { get; set; }
+        public Dictionary<string, string> Properties { get; set; }
+    }
+    public class Edge
+    {
+        public Edge()
+        {
+            Properties = new Dictionary<string, string>();
+        }
+        public Node SourceNode { get; set; }
+        public Node TargetNode { get; set; }
+        public EdgeType EdgeType { get; set; }
+        public string Identifier { get; set; }
+        public Dictionary<string, string> Properties { get; set; }
+    }
+
+    public enum NodeType
+    {
+        Project,
+        Class,
```

Comment:
We may want to have an `AbstractClass` value here too, we may run into issues with extraction treating them the same as non-abstract classes

Reply:
Good point,  will add  an issue for that. I don't think that's supported today